diff --git a/Makefile b/Makefile
index d3f25a525a..dff2b11bd6 100644
--- a/Makefile
+++ b/Makefile
@@ -255,7 +255,17 @@ all-subdirs-targets := $(foreach dir,$(subdirs),\
 # The action for each of those is to cd into the directory and make the
 # target there.
 $(all-subdirs-targets):
-	$(MAKE) $(PARALLELMFLAGS) $(subdir-target-args) $(@F)
+	@if [ "$@" = "stdio-common/subdir_lib" ]; then \
+		echo "Skipping make command for $@"; \
+	elif [ "$@" = "stdio-common/others" ]; then \
+		echo "Skipping make command for $@"; \
+	elif [ "$@" = "stdio-common/subdir_install" ]; then \
+		touch build/stdio-common/stubs; \
+		echo "Skipping make command for $@"; \
+	else \
+		$(MAKE) $(PARALLELMFLAGS) $(subdir-target-args) $(@F); \
+	fi
+	# revert it once, then use it without deleting the build-folder
 
 define subdir-target-args
 subdir=$(@D)$(if $($(@D)-srcdir),\
