diff --git a/.gitignore b/.gitignore
index 688980a5..891b527e 100644
--- a/.gitignore
+++ b/.gitignore
@@ -8,3 +8,17 @@ win64build
 
 ## qt-creator
 CMakeLists.txt.user*
+
+.suo
+*.log
+*.tlog
+*.obj
+*.sbr
+*.bsc
+*.ipch
+*.iobj
+*.ipdb
+*.recipe
+*.exp
+*.pdb
+*.pch
\ No newline at end of file
diff --git a/al/error.cpp b/al/error.cpp
index f852d5ed..75598ba7 100644
--- a/al/error.cpp
+++ b/al/error.cpp
@@ -97,7 +97,7 @@ void ALCcontext::setError(ALenum errorCode, const char *msg, ...)
         if(IsDebuggerPresent())
             DebugBreak();
 #elif defined(SIGTRAP)
-        raise(SIGTRAP);
+        //raise(SIGTRAP);
 #endif
     }
 
@@ -140,7 +140,7 @@ AL_API auto AL_APIENTRY alGetError() noexcept -> ALenum
         if(IsDebuggerPresent())
             DebugBreak();
 #elif defined(SIGTRAP)
-        raise(SIGTRAP);
+        //raise(SIGTRAP);
 #endif
     }
     return deferror;
diff --git a/alc/alc.cpp b/alc/alc.cpp
index 26ce9233..622793d4 100644
--- a/alc/alc.cpp
+++ b/alc/alc.cpp
@@ -1978,7 +1978,7 @@ void alcSetError(al::Device *device, ALCenum errorCode)
         if(IsDebuggerPresent())
             DebugBreak();
 #elif defined(SIGTRAP)
-        raise(SIGTRAP);
+        //raise(SIGTRAP);
 #endif
     }
 
diff --git a/alsoftrc.sample b/alsoftrc.sample
index 564e9f49..469329b5 100644
--- a/alsoftrc.sample
+++ b/alsoftrc.sample
@@ -189,7 +189,7 @@
 #            between 24 and 48 points, with anti-aliasing)
 #  fast_bsinc24 - same as bsinc24, except without interpolation between down-
 #                 sampling scales
-#resampler = gaussian
+#resampler = spline
 
 ## rt-prio: (global)
 #  Sets the real-time priority value for the mixing thread. Not all drivers may
diff --git a/common/althreads.h b/common/althreads.h
index d8491706..2283c08a 100644
--- a/common/althreads.h
+++ b/common/althreads.h
@@ -9,7 +9,7 @@
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 
-#elif defined(__APPLE__)
+#elif defined(__APPLE__) || 1
 
 #include <pthread.h>
 
@@ -79,7 +79,7 @@ public:
     [[nodiscard]]
     auto get() const noexcept -> T { return from_ptr(TlsGetValue(mTss)); }
 
-#elif defined(__APPLE__)
+#elif defined(__APPLE__) || 1
 
     pthread_key_t mTss{};
 
diff --git a/core/voice.h b/core/voice.h
index 9dbffdde..bb7c6e24 100644
--- a/core/voice.h
+++ b/core/voice.h
@@ -280,6 +280,6 @@ struct SIMDALIGN Voice {
     static void InitMixer(std::optional<std::string> resopt);
 };
 
-inline Resampler ResamplerDefault{Resampler::Gaussian};
+inline Resampler ResamplerDefault{Resampler::Spline};
 
 #endif /* CORE_VOICE_H */
