diff --git a/.gitignore b/.gitignore
index 688980a5..891b527e 100644
--- a/.gitignore
+++ b/.gitignore
@@ -8,3 +8,17 @@ win64build
 
 ## qt-creator
 CMakeLists.txt.user*
+
+.suo
+*.log
+*.tlog
+*.obj
+*.sbr
+*.bsc
+*.ipch
+*.iobj
+*.ipdb
+*.recipe
+*.exp
+*.pdb
+*.pch
\ No newline at end of file
diff --git a/al/error.cpp b/al/error.cpp
index 4d2015ce..f2428732 100644
--- a/al/error.cpp
+++ b/al/error.cpp
@@ -60,7 +60,7 @@ void ALCcontext::setErrorImpl(ALenum errorCode, const fmt::string_view fmt, fmt:
         if(IsDebuggerPresent())
             DebugBreak();
 #elif defined(SIGTRAP)
-        raise(SIGTRAP);
+        //raise(SIGTRAP);
 #endif
     }
 
@@ -114,7 +114,7 @@ AL_API auto AL_APIENTRY alGetError() noexcept -> ALenum
         if(IsDebuggerPresent())
             DebugBreak();
 #elif defined(SIGTRAP)
-        raise(SIGTRAP);
+        //raise(SIGTRAP);
 #endif
     }
     return deferror;
diff --git a/alc/alc.cpp b/alc/alc.cpp
index d378364b..503b0d2e 100644
--- a/alc/alc.cpp
+++ b/alc/alc.cpp
@@ -2004,7 +2004,7 @@ void alcSetError(al::Device *device, ALCenum errorCode)
         if(IsDebuggerPresent())
             DebugBreak();
 #elif defined(SIGTRAP)
-        raise(SIGTRAP);
+        //raise(SIGTRAP);
 #endif
     }
 
diff --git a/alsoftrc.sample b/alsoftrc.sample
index 166aae16..d5c46261 100644
--- a/alsoftrc.sample
+++ b/alsoftrc.sample
@@ -189,7 +189,7 @@
 #            between 24 and 48 points, with anti-aliasing)
 #  fast_bsinc24 - same as bsinc24, except without interpolation between down-
 #                 sampling scales
-#resampler = gaussian
+#resampler = spline
 
 ## rt-prio: (global)
 #  Sets the real-time priority value for the mixing thread. Not all drivers may
