diff --git a/include/opus_defines.h b/include/opus_defines.h
index cd8f4dde..cf8f0ff0 100644
--- a/include/opus_defines.h
+++ b/include/opus_defines.h
@@ -1,3 +1,6 @@
+// Modified version for Core Engine
+// Please use the original library from https://github.com/xiph/opus
+
 /* Copyright (c) 2010-2011 Xiph.Org Foundation, Skype Limited
    Written by Jean-Marc Valin and Koen Vos */
 /*
@@ -68,7 +71,7 @@ extern "C" {
 #  if defined(OPUS_BUILD) && defined(DLL_EXPORT)
 #   define OPUS_EXPORT __declspec(dllexport)
 #  else
-#   define OPUS_EXPORT
+#   define OPUS_EXPORT __declspec(dllimport)
 #  endif
 # elif defined(__GNUC__) && defined(OPUS_BUILD)
 #  define OPUS_EXPORT __attribute__ ((visibility ("default")))
@@ -77,6 +80,14 @@ extern "C" {
 # endif
 #endif
 
+#ifndef OPUS_CALL
+# if defined(_WIN32)
+#  define OPUS_CALL __cdecl
+# else
+#  define OPUS_CALL
+# endif
+#endif
+
 # if !defined(OPUS_GNUC_PREREQ)
 #  if defined(__GNUC__)&&defined(__GNUC_MINOR__)
 #   define OPUS_GNUC_PREREQ(_maj,_min) \
@@ -820,7 +831,7 @@ OPUS_EXPORT const char *opus_strerror(int error);
   *
   * @returns Version string
   */
-OPUS_EXPORT const char *opus_get_version_string(void);
+OPUS_EXPORT const char *OPUS_CALL opus_get_version_string(void);
 /**@}*/
 
 #ifdef __cplusplus
diff --git a/include/opus_types.h b/include/opus_types.h
index 7cf67558..35ed52ef 100644
--- a/include/opus_types.h
+++ b/include/opus_types.h
@@ -1,3 +1,6 @@
+// Modified version for Core Engine
+// Please use the original library from https://github.com/xiph/opus
+
 /* (C) COPYRIGHT 1994-2002 Xiph.Org Foundation */
 /* Modified by Jean-Marc Valin */
 /*
@@ -54,8 +57,8 @@
    typedef uint16_t opus_uint16;
    typedef int32_t opus_int32;
    typedef uint32_t opus_uint32;
-   typedef int64_t opus_int64;
-   typedef uint64_t opus_uint64;
+   typedef long long opus_int64;
+   typedef unsigned long long opus_uint64;
 #elif defined(_WIN32)
 
 #  if defined(__CYGWIN__)
