diff --git a/src/webp/decode.h b/src/webp/decode.h
index 0177b120..5c47cda1 100644
--- a/src/webp/decode.h
+++ b/src/webp/decode.h
@@ -36,6 +36,9 @@ typedef struct WebPDecoderConfig WebPDecoderConfig;
 
 // Return the decoder's version number, packed in hexadecimal using 8bits for
 // each of major/minor/revision. E.g: v2.5.7 is 0x020507.
+#if defined(_WIN32)
+__declspec(dllexport)
+#endif
 WEBP_EXTERN int WebPGetDecoderVersion(void);
 
 // Retrieve basic header information: width, height.
diff --git a/src/webp/demux.h b/src/webp/demux.h
index 846eeb15..bab93570 100644
--- a/src/webp/demux.h
+++ b/src/webp/demux.h
@@ -71,6 +71,9 @@ typedef struct WebPAnimDecoderOptions WebPAnimDecoderOptions;
 
 // Returns the version number of the demux library, packed in hexadecimal using
 // 8bits for each of major/minor/revision. E.g: v2.5.7 is 0x020507.
+#if defined(_WIN32)
+__declspec(dllexport)
+#endif
 WEBP_EXTERN int WebPGetDemuxVersion(void);
 
 //------------------------------------------------------------------------------
